### PR TITLE
pvr/GUIEPGGRidContainer: optimize appending to m_channelLayouts

### PR DESCRIPTION
**Pull request by MaxKellermann**
Eliminate temporary instance, and instead "emplace" to the container
and edit in-place.

### CODE REVIEW

File: xbmc/pvr/windows/GUIEPGGridContainer.cpp [hunk 1]
```diff
@@ -1567,55 +1567,49 @@ void CGUIEPGGridContainer::LoadLayout(TiXmlElement *layout)
   TiXmlElement *itemElement = layout->FirstChildElement("channellayout");
   while (itemElement)
   {
-    CGUIListItemLayout itemLayout;
-    itemLayout.LoadLayout(itemElement, GetParentID(), false, m_width, m_height);
-    m_channelLayouts.push_back(itemLayout);
+    m_channelLayouts.emplace_back();
```
Comment (fritsch) [inline xbmc/pvr/windows/GUIEPGGridContainer.cpp]: Seems something is missing here?

File: xbmc/pvr/windows/GUIEPGGridContainer.cpp [hunk 2]
```diff
@@ -1567,55 +1567,49 @@ void CGUIEPGGridContainer::LoadLayout(TiXmlElement *layout)
...
+    m_channelLayouts.back().LoadLayout(itemElement, GetParentID(), false, m_width, m_height);
     itemElement = itemElement->NextSiblingElement("channellayout");
   }
   itemElement = layout->FirstChildElement("focusedchannellayout");
   while (itemElement)
   {
-    CGUIListItemLayout itemLayout;
-    itemLayout.LoadLayout(itemElement, GetParentID(), true, m_width, m_height);
-    m_focusedChannelLayouts.push_back(itemLayout);
+    m_focusedChannelLayouts.emplace_back();
```
Comment (fritsch) [inline xbmc/pvr/windows/GUIEPGGridContainer.cpp]: Also here - parameter missing? c/p?

File: xbmc/pvr/windows/GUIEPGGridContainer.cpp (hunk 1 above)
Comment (MaxKellermann) [reply xbmc/pvr/windows/GUIEPGGridContainer.cpp]: I don't see that. It adds a new element, and invokes `LoadLayout`, just like the old code, only without the temporary. What is missing?

File: xbmc/pvr/windows/GUIEPGGridContainer.cpp (hunk 2 above)
Comment (MaxKellermann) [reply xbmc/pvr/windows/GUIEPGGridContainer.cpp]: Do you mean the `emplace_back` parameter? No, this invokes the default constructor, just like the old code did on the temporary (which is eliminated by this patch).

File: xbmc/pvr/windows/GUIEPGGridContainer.cpp (hunk 1 above)
Comment (MaxKellermann) [reply xbmc/pvr/windows/GUIEPGGridContainer.cpp]: You are right, it does call the default ctor; but there is still the `LoadLayout` call, identical to the old one, and it is needed.

**Review by ksooo — APPROVED** [review]
thanks.

**Pull request merged**